query/src/lib/is-fetching.ts

```diff
@@ -0,0 +1,35 @@
+import { QueryFilters, notifyManager } from '@tanstack/query-core';
+import { injectQueryClient } from './query-client';
+import { inject, Injectable, InjectionToken } from '@angular/core';
+import { distinctUntilChanged, Observable } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class IsFetchingService {
+  private queryClient = injectQueryClient();
+
+  use(filters?: QueryFilters) {
+    return new Observable<number>((observer) => {
+      observer.next(this.queryClient.isFetching(filters));
+      this.queryClient.getQueryCache().subscribe(
```

Comment:
We need to unsubscribe. I see that we didn't do that on v1 :(

Reply:
I think this could be also resolved then since im disposing the internal subscription on tearing down the observable